rules/apple_library.bzl

```diff
@@ -0,0 +1,673 @@
+"""Implementation of the `apple_library` rule."""
+
+load("//rules/framework:vfs_overlay.bzl", "framework_vfs_overlay", VFS_OVERLAY_FRAMEWORK_SEARCH_PATH = "FRAMEWORK_SEARCH_PATH")
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load("//rules/internal:objc_provider_utils.bzl", "objc_provider_utils")
+load("//rules:providers.bzl", "AvoidDepsInfo", "FrameworkInfo")
+load("//rules/framework:vfs_overlay.bzl", "VFSOverlayInfo", "make_vfsoverlay")
+load("@build_bazel_rules_swift//swift/internal:attrs.bzl", "swift_common_rule_attrs", "swift_deps_attr", "swift_toolchain_attrs")
+load(
+    "@build_bazel_rules_swift//swift/internal:build_settings.bzl",
+    "PerModuleSwiftCoptSettingInfo",
+    "additional_per_module_swiftcopts",
+)
+load(
+    "@build_bazel_rules_swift//swift/internal:compiling.bzl",
+    "apple_library_output_map",
+    "new_objc_provider",
+    "output_groups_from_other_compilation_outputs",
+)
+load(
+    "@build_bazel_rules_swift//swift/internal:feature_names.bzl",
+    "SWIFT_FEATURE_EMIT_SWIFTINTERFACE",
+    "SWIFT_FEATURE_ENABLE_LIBRARY_EVOLUTION",
+)
+
+#load("@build_bazel_rules_swift//swift/internal:linking.bzl", "create_linker_input")
+load("@build_bazel_rules_swift//swift/internal:providers.bzl", "SwiftInfo", "SwiftToolchainInfo")
+load("@build_bazel_rules_swift//swift/internal:swift_clang_module_aspect.bzl", "swift_clang_module_aspect")
```

Comment:
I think we can remove this once we get have a better story and approach to propagate PCMs. By nixing these aspects, we'll further reduce memory and speed up the analysis phase.